src/Solnet.Wallet/PublicKey.cs

```diff
@@ -79,7 +79,9 @@ public PublicKey(byte[] key)
         /// <param name="key">The public key as base58 encoded string.</param>
         public PublicKey(string key)
         {
-            Key = key ?? throw new ArgumentNullException(nameof(key));
+            if (key == null) throw new ArgumentNullException(nameof(key));
+            if (!FastCheck(key)) throw new ArgumentException("publickey contains a non-base58 character");

```

Comment:
Was thinking about adding a flag to check or ignore checking, with default to check?

```
public PublicKey(string key, bool validateKey = true)
{
            if (key == null) throw new ArgumentNullException(nameof(key));
            if (validateKey && !FastCheck(key)) throw new ArgumentException("publickey contains a non-base58 character");
 
 ....
}
```

Discuss.

Reply:
Eliminating any possibilities that a `PublicKey` coudl have been instantiated with an invalid key prevents a whole category of bugs in all code paths that use `PublicKey` instances. Allowing unsafe keys will necessetate duplicate checks in library code that accepts `PublicKey` instances to double-check whether it the value actually valid.

I cannot think of a single use case where you would want an instance of `PublicKey` that contains an invalid key. If there are, then perhaps a `PublicKeyUnchecked` derived class would better and a more explicit choice for developers. Any library code can use the `PublicKey` and be safe in the knowledge all instances have been pre-validated.

Reply:
![image](https://user-images.githubusercontent.com/87999215/176769196-7b7ce3dc-18a7-46b1-96f3-0f9df079d67c.png)

Reply:
You convinced me sir! A while ago we were doing decoding stuffs and we had to change the `PublicKey` class to prevent too much checks on the constructor (it was doing encoding back then). If perf is a problem, one could derive and use specifically yeah!